src/masked-input/masked-input.style.tsx

```diff
@@ -1,21 +1,41 @@
 import styled from "styled-components";
 import { Color } from "../color";
+import { InputGroup } from "../input-group";
+
+interface InputGroupWrapperProps {
+    readOnly: boolean;
+    isDisabled: boolean;
+}
 
 interface IconProps {
     $isDisabled?: boolean;
     $inactiveColor: string;
     $activeColor: string;
 }
 
+export const InputGroupWrapper = styled(InputGroup)<InputGroupWrapperProps>`
+    padding: 0 0 0 ${({ readOnly }) => (readOnly ? "0" : "1rem")};
+    input {
+        cursor: ${({ readOnly, isDisabled }) =>
+            readOnly && !isDisabled ? "pointer" : "initial"};
+    }
+`;
+
 export const IconContainer = styled.div<IconProps>`
     display: flex;
-    width: 1.125rem;
-    height: 1.125rem;
+    height: 3rem;
```

Comment:
note that the input group height is based on the content + border, so we need to `calc(3rem - 2px)` here to keep the overall component height at a consistent 48px

setting a fixed 3rem height on the `InputGroupWrapper` also works

Reply:
there's a requirement for the touch area of the mask icon to be of height 3rem and width 3.25rem... hmm if i do calc(3rem - 2px), the icon container height doesn't stretch to the entire input group height. any ideas how to resolve this?

Reply:
all good now!